ramp-database/ramp_database/tools/contributivity.py

```diff
@@ -0,0 +1,338 @@
+            return
+        combined_predictions_list.append(combined_predictions)
+        best_predictions_list.append(best_predictions)
+        combined_test_predictions_list.append(combined_test_predictions)
+        best_test_predictions_list.append(best_test_predictions)
+        test_is_list.append(cv_fold.test_is)
+
+    for submission in submissions:
+        submission.set_contributivity()
+    # if there are no predictions to combine, it crashed
+    combined_predictions_list = [c for c in combined_predictions_list
+                                 if c is not None]
+    if len(combined_predictions_list) > 0:
+        combined_predictions, scores = get_score_cv_bags(
+            score_type, combined_predictions_list,
+            ground_truths_train, test_is_list=test_is_list)
+        logger.info('Combined combined valid score = {}'.format(scores))
+        event.combined_combined_valid_score = float(scores[-1])
+    else:
+        event.combined_combined_valid_score = None
+
+    best_predictions_list = [c for c in best_predictions_list
+                             if c is not None]
+    if len(best_predictions_list) > 0:
+        _, scores = get_score_cv_bags(
+            score_type, best_predictions_list,
+            ground_truths_train, test_is_list=test_is_list)
+        logger.info('Combined foldwise best valid score = {}'.format(scores))
+        event.combined_foldwise_valid_score = float(scores[-1])
+    else:
+        event.combined_foldwise_valid_score = None
+
+    combined_test_predictions_list = [c for c in combined_test_predictions_list
+                                      if c is not None]
+    if len(combined_test_predictions_list) > 0:
+        combined_predictions, scores = get_score_cv_bags(
+            score_type, combined_test_predictions_list, ground_truths_test)
+        logger.info('Combined combined test score = {}'.format(scores))
+        event.combined_combined_test_score = float(scores[-1])
+    else:
+        event.combined_combined_test_score = None
+
+    best_test_predictions_list = [c for c in best_test_predictions_list
+                                  if c is not None]
+    if len(best_test_predictions_list) > 0:
+        _, scores = get_score_cv_bags(
+            score_type, best_test_predictions_list, ground_truths_test)
+        logger.info('Combined foldwise best valid score = {}'.format(scores))
+        event.combined_foldwise_test_score = float(scores[-1])
+    else:
+        event.combined_foldwise_test_score = None
+
+    session.commit()
+
+
+def _compute_contributivity_on_fold(session, cv_fold, ground_truths_valid,
+                                    start_time_stamp=None, end_time_stamp=None,
+                                    force_ensemble=False, min_improvement=0.0):
+    """Construct the best model combination on a single fold.
+
+    Using greedy forward selection with replacement. See
+    http://www.cs.cornell.edu/~caruana/ctp/ct.papers/
```

Comment:
This should be put as an hyperlink in a Reference section

Reply:
Done but please check, I'm not sure about the syntax.